src/VerifyExamples/Ast/Intermediate.elm

```diff
@@ -0,0 +1,210 @@
+module VerifyExamples.Ast.Intermediate
+    exposing
+        ( IntermediateAst
+        , fromString
+        , toAst
+        )
+
+import List.Extra
+import Maybe.Util exposing (oneOf)
+import Regex exposing (HowMany(..), Regex)
+import Regex.Util exposing (firstSubmatch, replaceAllWith, submatches)
+import String
+import String.Util exposing (unlines)
+import VerifyExamples.Ast as Ast exposing (Ast)
+
+
+type IntermediateAst
+    = ExpressionNotPrefixed String
+    | Expression Prefix String
+    | Function String String
+    | NewLine
+
+
+type Prefix
+    = ArrowPrefix
+    | ImportPrefix
+    | TypePrefix
+
+
+fromString : String -> List IntermediateAst
+fromString =
+    commentLines >> List.filterMap toIntermediateAst
+
+
+commentLines : String -> List String
+commentLines =
+    String.lines
+        >> List.concatMap splitOneLiners
+
+
+splitOneLiners : String -> List String
+splitOneLiners =
+    breakIntoTwoLines >> String.lines
+
+
+breakIntoTwoLines : String -> String
+breakIntoTwoLines a =
+    if Regex.contains expectationRegex a then
+        a
+    else
+        replaceAllWith arrowRegex "\n    --> " a
+
+
+toIntermediateAst : String -> Maybe IntermediateAst
+toIntermediateAst =
+    oneOf
+        [ firstSubmatch newLineRegex >> Maybe.map (always NewLine)
+        , firstSubmatch importRegex >> Maybe.map (Expression ImportPrefix)
+        , firstSubmatch typeRegex >> Maybe.map (Expression TypePrefix)
+        , firstSubmatch expectationRegex >> Maybe.map (Expression ArrowPrefix)
+        , firstSubmatch localFunctionRegex >> Maybe.map toFunctionExpression
+        , firstSubmatch assertionRegex >> Maybe.map ExpressionNotPrefixed
```

Comment:
:art: these could be refactored to be like `Regex -> (String -> IntermediateAst) -> Maybe IntermediateAst`